Item 9204: Sample type parent import alias inclusion in download template and display in details panel

cnathe opened the pull request:
#### Rationale
Sample type definitions allow for parent import aliases to be set to make it easier for users to map columns in the sample import files to the parent samples / sources. This set of PRs exposes those parent import aliases in the sample type details panel and download template column headers.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/581
* https://github.com/LabKey/platform/pull/2458
* https://github.com/LabKey/sampleManagement/pull/628
* https://github.com/LabKey/biologics/pull/941

#### Changes
* Add SampleTypeImportAliasRenderer and SourceTypeImportAliasRenderer to be used for sample type details panel rendering
* EntityInsertPanel update to get importAliases for target sample type to add to getTemplateUrl() params

cnathe committed:
Item 9204: SampleTypeImportAliasRenderer and SourceTypeImportAliasRenderer

cnathe committed:
v2.55.1-fb-sampleTypeImportAliasesUsage.0

cnathe committed:
Omit the import aliases columns from the sample type listing grid

cnathe committed:
v2.55.1-fb-sampleTypeImportAliasesUsage.1

cnathe committed:
EntityInsertPanel update to get importAliases for target sample type to add to getTemplateUrl() params

cnathe committed:
v2.55.1-fb-sampleTypeImportAliasesUsage.2

cnathe committed:
SampleTypeImportAliasRenderer change to display on same line instead of separate lines

cnathe committed:
v2.55.1-fb-sampleTypeImportAliasesUsage.3

cnathe committed:
Update release notes

cnathe committed:
Jest test update for two new renderers in DetailEditRenderer.tsx

cnathe committed:
Jest tests for ImportAliasRenderer

cnathe committed:
Merge branch 'develop' into fb_sampleTypeImportAliasesUsage

# Conflicts:
#	packages/components/package.json

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.0

cnathe committed:
Fix release notes ordering

labkey-susanh commented on packages/components/src/internal/components/entities/EntityInsertPanel.tsx:
```diff
@@ -311,6 +315,12 @@ export class EntityInsertPanelImpl extends Component<Props, StateProps> {
     gridInit = (insertModel: EntityIdCreationModel): void => {
         const schemaQuery = insertModel.getSchemaQuery();
         if (schemaQuery) {
+            getSampleTypeDetails(schemaQuery)
```
Should this be checking if the insertModel is for a sample type?  This panel can also used for data classes.

Ah, I guess the `getSampleTypeDetails` method is specific to sample types only in its default action.  A bit misleading, but functional.

cnathe replied:
Yeah, that loose connection between the `getSampleTypeDetails` and the `SampleSet` domain kind is a little odd for sure. 

I think adding a check for the insertModel type is a good idea. I'll do that.

labkey-susanh commented on packages/components/src/internal/renderers/SampleTypeImportAliasRenderer.tsx:
```diff
@@ -0,0 +1,59 @@
+}
+
+// export for jest testing
+export const ImportAliasRenderer: FC<RendererProps> = memo(props => {
```
I wonder if the file should be called `ImportAliasRenderer` since it contains both Sample and Source renderers.

cnathe replied:
Yes, good point. I started out with just the SampleTypeImportAliasRenderer, but then later realized I needed the source type one as well. I'll rename.

cnathe committed:
CR feedback
- rename file to ImportAliasRenderer.tsx
- EntityInsertPanel check for entityDataType of sample type before attempting to query importAliases (since they don't apply to sources)

cnathe committed:
Export the base component ImportAliasRenderer so that LKB can use it for the combined materialInputs and dataInputs renderer

cnathe committed:
Add jest test case for ImportAliasRenderer with multiple appRouteMap entries

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.1

cnathe committed:
Missed from previous commit - also omit the newly added 'ImportAliases' column

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.2

cnathe committed:
Fix check for import aliases query so that LKB override of SampleTypeDataType is accounted for

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.3

cnathe committed:
Update to display format of import alias key/value pairs

cnathe committed:
yarn run lint-branch-fix

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.4

cnathe committed:
Add check for updated display format of ImportAliasRenderer component

cnathe committed:
Sort import aliases by key in display

cnathe committed:
v2.56.0-fb-sampleTypeImportAliasesUsage.5

cnathe committed:
Merge branch 'develop' into fb_sampleTypeImportAliasesUsage

# Conflicts:
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
v2.56.1-fb-sampleTypeImportAliasesUsage.0

cnathe committed:
Fix for aliasMap split of route vs value

cnathe committed:
v2.56.1-fb-sampleTypeImportAliasesUsage.1

cnathe committed:
Merge branch 'develop' into fb_sampleTypeImportAliasesUsage

# Conflicts:
#	packages/components/package.json

cnathe committed:
Update release notes date and version

cnathe committed:
yarn run lint-branch-fix

cnathe committed:
v2.57.0
